src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -737,3 +739,35 @@ fn plan_kafka_security(
 
     Ok((tls, sasl))
 }
+
+fn is_valid_az_format(az: &str) -> bool {
+    let az_pattern = Regex::new(r"^[a-z]{3,4}\d-az[1-6]$").unwrap();
+    az_pattern.is_match(az)
+}
```

Comment:
It worries me a bit injecting a pattern when we don't know what rules AWS uses internally for naming these, but this is probably as good as we can do without injecting the complete list.